pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java

```diff
@@ -242,6 +244,7 @@ public static class Params {
       private long _memoryUsedBytes;
       private long _segmentSizeBytes;
       private String _streamPartitionMsgOffset;
+      private boolean _pauselessConsumptionEnabled;
```

Comment:
Instead of changing the existing protocol, please add a new controller endpoint that is called from the server when it needs the idealstate modified before segment upload.

This will avoid a lot of `if pauselessEnabled` statements in the code path